### PR TITLE
Simple Button Tool

### PR DESCRIPTION
**Pull request by corranwebster**
Basically a common tool that can be used whenever we want something to be "clickable" or "togglable".  Probably of significant utility in Chaco.

Has tests and documentation.

Ready for review.

### CODE REVIEW

File: docs/source/enable_basic_tools.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+To get checkbox-style behaviour, set :py:attr:`togglable` to ``True`` and
+then every click will invert the :py:attr:`checked` trait.  The toggle state
+can also be  changed via the :py:meth:`toggle` method, which does not fire the
+:py:attr:`clicked` event when called.  For buttons with multi-state toggles,
+subclasses can override the :py:meth:`toggle` method to perform more complex
+state changes.
+
+By default, the tool responds to clicks that are within the associated
+component, but subclasses can override this behaviour by replacing the
+:py:method:``is_clickable`` method with something else.
```
Comment (jwiggins) [inline docs/source/enable_basic_tools.rst]: Any reason for the double backticks here?

File: docs/source/enable_basic_tools.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+HoverTool
+~~~~~~~~~
+
+The :py:class:`HoverTool` is a simple tool that calls a callback when the
+mouse has been held steadily over the component for a period of time.
+
+MoveTool
+~~~~~~~~
+
+A :py:class:`DragTool` subclass that allows a user to move a component around
```
Comment (jwiggins) [inline docs/source/enable_basic_tools.rst]: Copypasta

File: docs/source/enable_basic_tools.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+its container by dragging.
+
+ResizeTool
+~~~~~~~~~~
+
+A :py:class:`DragTool` subclass that allows a user to resize a component by
```
Comment (jwiggins) [inline docs/source/enable_basic_tools.rst]: Copypasta

File: docs/source/enable_basic_tools.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+dragging from the edges of the component.
+
+ValueDragTool
+~~~~~~~~~~~~~
+
+A :py:class:`DragTool` that allows a drag operation to set an arbitrary
```
Comment (jwiggins) [inline docs/source/enable_basic_tools.rst]: Copypasta

Comment (jwiggins) [inline docs/source/enable_basic_tools.rst]: Actually, I think maybe "subclass" is missing

File: enable/tests/tools/button_tool_test_case.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+                self.assertTraitChanges(tool, 'checked'):
+            tool.toggle()
+
+        self.assertTrue(tool.checked)
+
+    def test_toggle_function(self):
+        tool = self.tool
+        tool.togglable = True
+
+        with self.assertTraitDoesNotChange(tool, 'down', count=1), \
```
Comment (jwiggins) [inline enable/tests/tools/button_tool_test_case.py]: Isn't the `count=1` ignored here?

File: enable/tests/tools/button_tool_test_case.py [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+            tool.toggle()
+
+        self.assertTrue(tool.checked)
+
+    def test_toggle_function_checked(self):
+        tool = self.tool
+        tool.togglable = True
+        tool.checked = True
+
+        with self.assertTraitDoesNotChange(tool, 'down', count=1), \
```
Comment (jwiggins) [inline enable/tests/tools/button_tool_test_case.py]: ditto

File: enable/tools/button_tool.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+    """
+    #-------------------------------------------------------------------------
+    # 'ButtonTool' interface
+    #-------------------------------------------------------------------------
+
+    #: Event fired when button is clicked
+    clicked = Event
+
+    #: Is the button togglable?
```
Comment (jwiggins) [inline enable/tools/button_tool.py]: togglable -> toggled?

**Comment by jwiggins** [comment]
Just a few minor nitpicks. Other than that this looks nice.

**Comment by corranwebster** [comment]
Took me a while to have the spare cycles to look at this, but I've fixed issues.

However it looks like we are getting failures in the build due to something wrong with PIL.  I think they are unrelated.

**Comment by jwiggins** [comment]
Thanks for the updates! I'm not going to count the Travis failures against this PR.

**Pull request merged**